src/Discovery/PathFinderCore.php

```diff
@@ -17,19 +17,18 @@ class PathFinderCore extends PathFinderBase implements PathFinderInterface {
   public function find($seed) {
     $seed = realpath($seed);
     // Try to create the iterator with the seed.
-    try {
+    $directory = NULL;
+    if (is_dir($seed)) {
       $directory = new \DirectoryIterator($seed);
     }
-    catch (\RuntimeException $e) {
-      // If the seed was not a directory, then get the parent directory.
-      $path_info = pathinfo($seed);
-      $directory = new \DirectoryIterator($path_info['dirname']);
+    else {
+      $directory = new \DirectoryIterator(dirname($seed));
     }
     // Starting at the directory containing the seed path, we go one directory
     // up and up and up until we reach the Drupal root.
     do {
       if ($this->isDrupalRoot($directory)) {
-        return $this->cleanDirPath($directory->getPathName()) . $this->path;
+        return dirname($this->cleanDirPath($directory->getPathName())) . $this->path;
```

Comment:
Not sure if that is a good idea though. The code behaves differently when the path does not end in `/.`.